analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,254 @@
+    #TODO check for values
+    standardFusioncalls <- standardFusioncalls[-grep(artifactFilter,standardFusioncalls$annots),]
+  }
+
+  if( !missing(junctionReadCountFilter ) ){
+    standardFusioncalls <- standardFusioncalls[which(standardFusioncalls$JunctionReadCount>=junctionReadCountFilter),]
+  }
+
+  if( !missing(spanningFragCountFilter ) ){
+    # false positive calls either at the start or end of a transcript will be supported by an uneven majority of support from spanning fragments compared to junction reads
+    # to remove these calls we are implementing this condition below
+    standardFusioncalls <- standardFusioncalls[which( (standardFusioncalls$SpanningFragCount - standardFusioncalls$JunctionReadCount ) <= spanningFragCountFilter),]
+  }
+  
+  return(standardFusioncalls)
+}
+
+# QC filter: artifact and read support
+QCFiltered<-fusion_filtering_QC(standardFusioncalls = standardFusioncalls,junctionReadCountFilter = junctionReadCountFilter,spanningFragCountFilter = spanningFragCountFilter,readingFrameFilter = readingFrameFilter,artifactFilter = artifactFilter,reathroughFilter = reathroughFilter)
+
+############################################
+
+########### Expression filtering ###########
+
+# load expressionMatrix RDS for expression based filtering for less than given threshold
+expressionMatrix<-readRDS(expressionMatrix)
+expressionMatrix <- cbind(expressionMatrix, colsplit(expressionMatrix$gene_id, pattern = '_', names = c("EnsembleID","GeneSymbol")))
+
+
+# The idea from @jaclyn-taroni 
+# Generate two data frames that keep track of all gene symbols involved for each sample-fusion name pair and contain a sample's expression value for a gene symbol in long format. Use these to  filter all the available fusions to just the ones with either gene from the fusion pair to have expression above the threshold.
+
+fusion_sample_gene_df <- QCFiltered %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample, FusionName, Gene1A, Gene1B, Gene2A, Gene2B) %>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = GeneSymbol) %>%
+  # Get rid of the Gene1A, Gene1B, Gene2A, Gene2B information              
+  dplyr::select(-gene_position) %>%
+  # Remove columns without gene symbols
+  dplyr::filter(GeneSymbol != "") %>%
+  # This is for illustrations sake, only
+  dplyr::arrange(Sample, FusionName) %>%
+  # Retain only distinct rows
+  dplyr::distinct()
+
+
+expression_long_df <- expressionMatrix %>%
+  # Keep the gene symbols and the samples themselves
+  dplyr::select(GeneSymbol, dplyr::starts_with("BS_")) %>%
+  # Get the data into long format
+  reshape2::melt(variable.name = "Sample",
+                 value.name = "expression_value") %>%
+  # Remove rows with expression that is too low
+  dplyr::filter(expression_value > expressionFilter)
+
+
```

Comment:
This solution doesn't current include a warning, you could put something like the following here:

```r
if (!all(fusion_sample_gene_df$Sample %in% expression_long_df$Sample)) {
  warning("Not all samples in expression file. Only returning fusions for samples in expressionMatrix.")
}
```